url9d04094941_petermikitsh_PizzaDeliverySystem_tgz-pJ8-testdrivers_ManagerReportTestDriverJ8/src/ninja/Kitchen.java

```diff
@@ -0,0 +1,549 @@
+	}
+
+	/**
+	 * Gets the list of oven runners.
+	 */
+	public static OvenRunner getOvenRunner() {
+		return _ovenRunner;
+	}
+
+	/**
+	 * Gets the list of all orders currently in the kitchen.
+	 */
+	public static Vector<Order> getOrders() {
+		return _allOrders;
+	}
+
+	/**
+	 * Adds an order to the kitchen system.
+	 * 
+	 * @param order
+	 *            The new order.
+	 */
+	public static void addOrder(Order order) {
+
+		// Mark the order start time.
+		order.setTimeOrdered( SystemTime.getTime() );
+		
+		// Add the order to the list of orders.
+		_allOrders.add(order);
+
+		// Send the order to be broken into food items.
+		Kitchen.decomposeOrder(order);
+
+	}
+
+	/**
+	 * Breaks an order into constituent FIs and puts them in the right place.
+	 * 
+	 * @param order
+	 */
+	private static void decomposeOrder(Order order) {
+
+		// For now, just put all of the food items in the awaiting prep
+		// list.
+		for (FoodItem item : order.getFoodItems()) {
+
+			// Set the status of the food item "Awaiting Prep", time stamp it,
+			// and add it to the queue.
+			item.setStatus(FoodItemStatus.AwaitingPreparation);
+			_awaitingPrepFIs.add(item);
+
+		}
+
+	}
+
+	/**
+	 * Removes an order from the kitchen.
+	 * 
+	 * @param order
+	 *            The order to remove.
+	 */
+	public static void removeOrder(Order order) {
+
+		// Remove from the all orders list.
+		_allOrders.remove(order);
+
+		// Also remove each food item from the awaiting prep list.
+		for (FoodItem item : order.getFoodItems()) {
+
+			_awaitingPrepFIs.remove(item);
+
+		}
+
+	}
+	
+	/**
+	 * Sends a food item to the awaiting oven list.
+	 * 
+	 * @param item
+	 *            The FI to send.
+	 */
+	public static void sendToAwaitingOven(FoodItem item) {
+
+		// Add to the list.
+		item.setStatus(FoodItemStatus.AwaitingOven);
+		_awaitingOvenFIs.add(item);
+
+	}
+
+	/**
+	 * Sends a food item to the given oven.
+	 * 
+	 * @param item
+	 *            The FI to send.
+	 * @param oven
+	 *            The oven to add to.
+	 */
+	public static void sendToOven(FoodItem item, Oven oven) {
+
+		// Add the food item to the oven, and set a time stamp on the food item.
+		item.setStatus(FoodItemStatus.Cooking);
+		oven.addFoodItem(item);
+
+	}
+
+	/**
+	 * Sends a food item to the awaiting delivery list. (Actually just marks the
+	 * food item as "awaiting delivery".
+	 * 
+	 * @param item
+	 *            The FI to send.
+	 */
+	public synchronized static void sendToWarmingArea(FoodItem item) {
+
+		// Set a time stamp and mark the item as awaiting delivery.
+		item.setStatus(FoodItemStatus.AwaitingDelivery);
+
+	}
+
+	/**
+	 * Called when a cook wants a food item.
+	 * 
+	 * @return A food item if there is one awaiting prep, or null if not.
+	 */
+	public synchronized static FoodItem cookRetrieve() {
+
+		FoodItem retItem = null;
+
+		if (!_awaitingPrepFIs.isEmpty()) {
+
+			retItem = _awaitingPrepFIs.firstElement();
+			_awaitingPrepFIs.remove(retItem);
+
+		}
+
+		return retItem;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 405:414, 425:433.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.